pkg/apis/garden/validation/validation.go

```diff
@@ -1777,17 +1777,27 @@ func validateKubeControllerManager(kubernetesVersion string, kcm *garden.KubeCon
 	return allErrs
 }
 
-func validateKubeProxy(kp *garden.KubeProxyConfig, fldPath *field.Path) field.ErrorList {
+func validateKubeProxy(kp *garden.KubeProxyConfig, proxyMode garden.ProxyMode, version string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
-
 	if kp != nil {
 		if kp.Mode == nil {
 			allErrs = append(allErrs, field.Required(fldPath.Child("mode"), "must be set when .spec.kubernetes.kubeProxy is set"))
-		} else if mode := *kp.Mode; !availableProxyMode.Has(string(mode)) {
-			allErrs = append(allErrs, field.NotSupported(fldPath.Child("mode"), mode, availableProxyMode.List()))
+		} else if proxyMode := *kp.Mode; !availableProxyMode.Has(string(proxyMode)) {
```

Comment:
Please try to avoid shadowing of `proxyMode` here. Rather stick to `mode` like before.